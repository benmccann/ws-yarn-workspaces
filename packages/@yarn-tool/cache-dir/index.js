"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getYarnCacheDir = void 0;
const upath2_1 = require("upath2");
const fs_extra_1 = require("fs-extra");
const cross_spawn_extra_1 = require("cross-spawn-extra");
function getYarnCacheDir() {
    try {
        let cp = cross_spawn_extra_1.sync('yarn', [
            'config',
            'current',
            '--json',
        ], {
            stripAnsi: true,
            env: process.env,
        });
        let data = JSON.parse(JSON.parse(cp.stdout.toString()).data);
        if (data.tempFolder) {
            return upath2_1.normalize(data.tempFolder);
        }
    }
    catch (e) {
    }
    if (process.env.YARN_CACHE_FOLDER && fs_extra_1.pathExistsSync(process.env.YARN_CACHE_FOLDER)) {
        return upath2_1.normalize(process.env.YARN_CACHE_FOLDER);
    }
}
exports.getYarnCacheDir = getYarnCacheDir;
exports.default = getYarnCacheDir;
//# sourceMappingURL=index.js.map