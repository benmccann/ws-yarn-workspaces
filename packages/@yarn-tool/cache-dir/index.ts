import { join, normalize } from "upath2";
import { pathExistsSync } from 'fs-extra';
import { sync as crossSpawnExtra } from 'cross-spawn-extra';

export function getYarnCacheDir(): string
{
	try
	{
		let cp = crossSpawnExtra('yarn', [
			'config',
			'current',
			'--json',
		], {
			stripAnsi: true,
			env: process.env,
		});

		let data = JSON.parse(JSON.parse(cp.stdout.toString()).data);

		if (data.tempFolder)
		{
			return normalize(data.tempFolder)
		}
	}
	catch (e)
	{

	}

	if (process.env.YARN_CACHE_FOLDER && pathExistsSync(process.env.YARN_CACHE_FOLDER))
	{
		return normalize(process.env.YARN_CACHE_FOLDER)
	}
}

export default getYarnCacheDir
