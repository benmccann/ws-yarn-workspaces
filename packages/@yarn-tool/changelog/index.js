"use strict";
/**
 * Created by user on 2020/6/15.
 */
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
tslib_1.__exportStar(require("./lib/types"), exports);
tslib_1.__exportStar(require("./lib/lerna"), exports);
tslib_1.__exportStar(require("./lib/yargs-setting"), exports);
const lerna_1 = require("./lib/lerna");
exports.default = lerna_1.updateChangelogByCwd;
//# sourceMappingURL=index.js.map