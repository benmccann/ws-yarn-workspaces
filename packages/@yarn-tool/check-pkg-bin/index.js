"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const pkg_1 = require("./lib/pkg");
(0, tslib_1.__exportStar)(require("./lib/core"), exports);
(0, tslib_1.__exportStar)(require("./lib/ws"), exports);
(0, tslib_1.__exportStar)(require("./lib/pkg"), exports);
exports.default = pkg_1.checkPkgDir;
//# sourceMappingURL=index.js.map