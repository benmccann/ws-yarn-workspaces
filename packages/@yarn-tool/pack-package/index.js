"use strict";
/**
 * Created by user on 2020/4/9.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.packPackage = exports.getPackedTarballInfo = exports.printPackedTarballInfo = exports.packTargetPackage = void 0;
const pack_1 = require("./lib/pack");
const info_1 = require("./lib/info");
const print_1 = require("./lib/print");
var pack_2 = require("./lib/pack");
Object.defineProperty(exports, "packTargetPackage", { enumerable: true, get: function () { return pack_2.packTargetPackage; } });
var print_2 = require("./lib/print");
Object.defineProperty(exports, "printPackedTarballInfo", { enumerable: true, get: function () { return print_2.printPackedTarballInfo; } });
var info_2 = require("./lib/info");
Object.defineProperty(exports, "getPackedTarballInfo", { enumerable: true, get: function () { return info_2.getPackedTarballInfo; } });
function packPackage(options) {
    return pack_1.packTargetPackage(options)
        .then(async (data) => {
        let tarball;
        const object = Object.freeze({
            ...data,
            async tarball() {
                return tarball !== null && tarball !== void 0 ? tarball : (tarball = await info_1.getPackedTarballInfo(data));
            },
            async log() {
                return print_1.printPackedTarballInfo(await object.tarball());
            },
        });
        return object;
    });
}
exports.packPackage = packPackage;
exports.default = packPackage;
//# sourceMappingURL=index.js.map