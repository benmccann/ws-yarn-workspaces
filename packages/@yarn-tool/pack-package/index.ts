/**
 * Created by user on 2020/4/9.
 */

import { IPackageJson } from '@ts-type/package-dts';
import { packTargetPackage } from './lib/pack';
import { getPackedTarballInfo } from './lib/info';
import { printPackedTarballInfo } from './lib/print';
import { IPackedTarballInfo } from './lib/types';

export { packTargetPackage } from './lib/pack';

export { printPackedTarballInfo } from './lib/print';
export { getPackedTarballInfo } from './lib/info';

export function packPackage(options: {
	pkg?: IPackageJson,
	packageDir: string,
	packageTarball?: string,
	versionPrefix?: string,
})
{
	return packTargetPackage(options)
		.then(async (data) => {
			let tarball: IPackedTarballInfo;

			const object = Object.freeze({
				...data,
				async tarball()
				{
					return tarball ?? (tarball = await getPackedTarballInfo(data))
				},
				async log()
				{
					return printPackedTarballInfo(await object.tarball())
				},
			});

			return object
		})
}

export default packPackage
