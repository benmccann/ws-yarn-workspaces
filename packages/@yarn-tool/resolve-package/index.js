"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const upath2_1 = require("upath2");
function resolvePackage(name, options) {
    let pkgRoot = upath2_1.dirname(require.resolve(`${name}/package.json`, options));
    return {
        name,
        pkgRoot,
        pkg: require(`${pkgRoot}/package.json`),
    };
}
exports.resolvePackage = resolvePackage;
exports.default = resolvePackage;
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyJpbmRleC50cyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiOztBQUFBLG1DQUFpQztBQUdqQyxTQUFnQixjQUFjLENBQUMsSUFBWSxFQUFFLE9BQStCO0lBRTNFLElBQUksT0FBTyxHQUFHLGdCQUFPLENBQUMsT0FBTyxDQUFDLE9BQU8sQ0FBQyxHQUFHLElBQUksZUFBZSxFQUFFLE9BQU8sQ0FBQyxDQUFDLENBQUM7SUFFeEUsT0FBTztRQUNOLElBQUk7UUFDSixPQUFPO1FBQ1AsR0FBRyxFQUFFLE9BQU8sQ0FBQyxHQUFHLE9BQU8sZUFBZSxDQUF3QjtLQUM5RCxDQUFBO0FBQ0YsQ0FBQztBQVRELHdDQVNDO0FBRUQsa0JBQWUsY0FBYyxDQUFBIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IHsgZGlybmFtZSB9IGZyb20gJ3VwYXRoMic7XG5pbXBvcnQgdHlwZSB7IElQYWNrYWdlSnNvbiB9IGZyb20gJ0B0cy10eXBlL3BhY2thZ2UtZHRzJztcblxuZXhwb3J0IGZ1bmN0aW9uIHJlc29sdmVQYWNrYWdlKG5hbWU6IHN0cmluZywgb3B0aW9ucz86IHsgcGF0aHM/OiBzdHJpbmdbXTsgfSlcbntcblx0bGV0IHBrZ1Jvb3QgPSBkaXJuYW1lKHJlcXVpcmUucmVzb2x2ZShgJHtuYW1lfS9wYWNrYWdlLmpzb25gLCBvcHRpb25zKSk7XG5cblx0cmV0dXJuIHtcblx0XHRuYW1lLFxuXHRcdHBrZ1Jvb3QsXG5cdFx0cGtnOiByZXF1aXJlKGAke3BrZ1Jvb3R9L3BhY2thZ2UuanNvbmApIGFzIGFueSBhcyBJUGFja2FnZUpzb24sXG5cdH1cbn1cblxuZXhwb3J0IGRlZmF1bHQgcmVzb2x2ZVBhY2thZ2VcbiJdfQ==