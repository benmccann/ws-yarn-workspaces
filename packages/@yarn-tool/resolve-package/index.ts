import { dirname, normalize, sep, resolve } from 'upath2';
import type { IPackageJson } from '@ts-type/package-dts';
import { sync as pkgDir } from 'pkg-dir';

/**
 * @see RequireResolve
 */
export interface IOptions
{
	paths?: string[];
}

export function resolvePackageCore(moduleName: string, options?: IOptions)
{
	const entryPointLocation = require.resolve(moduleName, options);
	const pkgRoot = pkgDir(entryPointLocation);

	return {
		name: moduleName,
		pkgRoot,
		entryPointLocation,
	}
}

export function resolvePackageRoot(moduleName: string, options?: IOptions)
{
	return resolvePackageCore(moduleName, options).pkgRoot
}

export function resolvePackageJsonLocation(moduleName: string, options?: IOptions)
{
	return resolve(resolvePackageCore(moduleName, options).pkgRoot, 'package.json')
}

export function createResolveLocationFn(moduleName: string, options?: IOptions)
{
	const { pkgRoot } = resolvePackageCore(moduleName, options);
	return (path: string, ...paths: string[]) => resolve(pkgRoot, path, ...paths)
}

export function readModulePackageJson<P extends IPackageJson>(moduleName: string, options?: IOptions): P
{
	return require(resolvePackageJsonLocation(moduleName, options))
}

export function resolvePackage<P extends IPackageJson>(moduleName: string, options?: IOptions)
{
	const _ = resolvePackageCore(moduleName, options);
	const { pkgRoot } = _;

	const pkgJsonLocation: string = resolve(pkgRoot, 'package.json');

	return {
		..._,
		pkg: require(pkgJsonLocation) as any as P,
		pkgJsonLocation,
		resolveLocation(path: string, ...paths: string[])
		{
			return resolve(pkgRoot, path, ...paths)
		},
	}
}

export default resolvePackage
