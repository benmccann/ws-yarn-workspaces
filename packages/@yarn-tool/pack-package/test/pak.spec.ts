import { basename, dirname, extname } from 'path';
import { packPackage } from '../index';
import { resolvePackage } from '@yarn-tool/resolve-package';

test(`packPackage`, async () =>
{
	const packageDir = resolvePackage('@yarn-tool/script-lifecycle/test/res/pkg-a').pkgRoot;

	let actual = await packPackage({
		packageDir,
	});

	expect(actual).toMatchSnapshot({
		packageDir: expect.any(String),
		packageTarball: expect.any(String),
		pkg: expect.any(Object),
	});

	await actual.log();

	console.dir(await actual.tarball());

	expect(await actual.tarball()).toMatchSnapshot({
		tarFilePath: expect.any(String),
	});

});
