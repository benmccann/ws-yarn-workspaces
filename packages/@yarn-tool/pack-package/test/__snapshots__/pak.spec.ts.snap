// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`packPackage 1`] = `
Object {
  "log": [Function],
  "packageDir": Any<String>,
  "packageTarball": Any<String>,
  "pkg": Any<Object>,
  "tarball": [Function],
}
`;
